packages/beacon-state-transition/src/cache/epochContext.ts

```diff
@@ -103,6 +103,17 @@ export class EpochContext {
    * 32 x Number
    */
   proposers: ValidatorIndex[];
+
+  /**
+   * Map of Indexes of the block proposers keyed by the next epoch.
+   *
+   * We allow requesting proposal duties only one epoch in the future
+   * Note: There is a small probability that returned validators differs
+   * than what is returned when the epoch is reached.
+   *
+   * 32 x Number
+   */
+  nextEpochProposers: Map<Epoch, ValidatorIndex[]> = new Map<Epoch, ValidatorIndex[]>();
```

Comment:
Cache exactly 1 or 0 next proposers for the next epoch of this epochCtx. We should research the safety of +2, +3 epoch duties before generalizing.

Reply:
> Cache exactly 1 or 0 next proposers for the next epoch of this epochCtx

Yeah. That is what is being done. Are you hinting at something else?

Reply:
If there's exactly 1 possible value for that why do a map?

```ts
nextEpochProposers: ValidatorIndex[] | null;
```
is more intuitive